### ISSUE TITLE
Incompatible base classes reported for generic base classes

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#5791]
The following code
```python
from typing import Generic, TypeVar

EP = TypeVar('EP')
NP = TypeVar('NP')

class NaturalParametrization(Generic[EP]):
    pass

class HasEntropyEP:
    pass

class BernoulliEP(HasEntropyEP):
    pass

class HasEntropyNP(NaturalParametrization[HasEntropyEP]):
    pass

class BernoulliNP(HasEntropyNP,
                  NaturalParametrization[BernoulliEP]):
    pass
```
gives
```
/home/neil/src/efax/a.py
  /home/neil/src/efax/a.py:18:7 - error: Base classes of BernoulliNP are mutually incompatible
    Base class "NaturalParametrization[BernoulliEP]" is incompatible with type "NaturalParametrization[HasEntropyEP]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
```
If I'm not mistaken, then in an ideal world, it would be nice for Pyright to infer the base class: `NaturalParametrization[BernoulliEP & HasEntropyEP]`, which is just `NaturalParametrization[BernoulliEP]` since `BernoulliEP < HasEntropyEP`.
